apsuite/optics_analysis/base_correction.py

```diff
@@ -0,0 +1,188 @@
+"""."""
+
+from collections import namedtuple as _namedtuple
+from copy import deepcopy as _dcopy
+import numpy as np
+
+__KNOBTYPES = _namedtuple('KnobTypes', ['Focusing', 'Defocusing'])
+
+
+class KnobTypes(__KNOBTYPES):
+    """."""
+
+    @property
+    def ALL(self):
+        """."""
+        return self.Focusing + self.Defocusing
+
+
+class BaseCorr():
+    """."""
+
+    METHODS = _namedtuple('Methods', ['Additional', 'Proportional'])(0, 1)
+    GROUPING = _namedtuple('Grouping', ['Individual', 'TwoKnobs'])(0, 1)
+    CORR_STATUS = _namedtuple('CorrStatus', ['Fail', 'Sucess'])(0, 1)
+
+    def __init__(self, model=None, acc=None, method=None, grouping=None):
+        """."""
+        self.model = model
+        self.acc = acc
+        self._method = BaseCorr.METHODS.Proportional
+        self._grouping = BaseCorr.GROUPING.TwoKnobs
+        self.knobs = None
+        self.fam = None
+        self.strength_type = None
+        self.method = method
+        self.grouping = grouping
+
+    @property
+    def grouping(self):
+        """."""
+        return self._grouping
+
+    @grouping.setter
+    def grouping(self, value):
+        if value is None:
+            return
+        elif isinstance(value, str):
+            self._grouping = int(value in BaseCorr.GROUPING._fields[1])
+        elif int(value) in BaseCorr.GROUPING:
+            self._grouping = int(value)
+
+    @property
+    def grouping_str(self):
+        """."""
+        return BaseCorr.GROUPING._fields[self._grouping]
+
+    @property
+    def method(self):
+        """."""
+        return self._method
+
+    @method.setter
+    def method(self, value):
+        if value is None:
+            return
+        elif isinstance(value, str):
+            self._method = int(value in BaseCorr.METHODS._fields[1])
+        elif int(value) in BaseCorr.METHODS:
+            self._method = int(value)
+
+    @property
+    def method_str(self):
+        """."""
+        return BaseCorr.METHODS._fields[self._method]
+
+    def define_knobs(self, f_knobs, d_knobs, strength_type):
```

Comment:
With the suggestion made above, regarding the attribute `strength_type`, I think this method is not necessary. The knobs can be defined in the `__init__` method of the derived class.